ci-operator/step-registry/assisted/baremetal/operator/catalog-publish/assisted-baremetal-operator-catalog-publish-commands.sh

```diff
@@ -0,0 +1,100 @@
+echo "## Generate and publish bundle to ${BUNDLE_DESTINATION}"
+
+SRC_DIR="${PWD}"
+
+tmp=$(mktemp -d)
+cd "${tmp}"
+
+# Build the bundle image from an empty base and set the required labels
+# It does the same as https://github.com/openshift/assisted-service/blob/master/deploy/olm-catalog/bundle.Dockerfile
+umoci init --layout bundle
+umoci new --image bundle:latest
+umoci config --config.label 'operators.operatorframework.io.bundle.mediatype.v1=registry+v1' --image bundle:latest
+umoci config --config.label 'operators.operatorframework.io.bundle.mediatype.v1=registry+v1' --image bundle:latest
+umoci config --config.label 'operators.operatorframework.io.bundle.manifests.v1=manifests/' --image bundle:latest
+umoci config --config.label 'operators.operatorframework.io.bundle.metadata.v1=metadata/' --image bundle:latest
+umoci config --config.label 'operators.operatorframework.io.bundle.package.v1=assisted-service-operator' --image bundle:latest
+umoci config --config.label 'operators.operatorframework.io.bundle.channels.v1=alpha,ocm-2.5' --image bundle:latest
+umoci config --config.label 'operators.operatorframework.io.bundle.channel.default.v1=alpha' --image bundle:latest
+umoci config --config.label 'operators.operatorframework.io.metrics.builder=operator-sdk-v1.10.0+git' --image bundle:latest
+umoci config --config.label 'operators.operatorframework.io.metrics.mediatype.v1=metrics+v1' --image bundle:latest
+umoci config --config.label 'operators.operatorframework.io.metrics.project_layout=go.kubebuilder.io/v3' --image bundle:latest
+umoci unpack --rootless --image bundle:latest bundle
+
+# Update the manifests to points to the digest of the images it contains
+python3 "${SRC_DIR}/deploy/olm-catalog/pin-latest.py"
+cp -r "${SRC_DIR}/deploy/olm-catalog/manifests" bundle/rootfs/
+cp -r "${SRC_DIR}/deploy/olm-catalog/metadata" bundle/rootfs/
+
+# Publish the bundle
+umoci repack --image bundle:latest bundle
+skopeo copy oci:bundle:latest "docker://${BUNDLE_DESTINATION}"
+
+# Get bundle digest
+BUNDLE_DIGEST=$(skopeo inspect oci:bundle:latest | jq -r '.Digest')
+BUNDLE_DESTINATION_DIGEST="${REGISTRY_HOST}/${REGISTRY_ORG}/${REGISTRY_BUNDLE_REPOSITORY_NAME}@${BUNDLE_DIGEST}"
+
+CATALOG_DESTINATION="${REGISTRY_HOST}/${REGISTRY_ORG}/${REGISTRY_CATALOG_REPOSITORY_NAME}:${REGISTRY_CATALOG_REPOSITORY_TAG}"
+echo "## Generate and publish catalog to ${CATALOG_DESTINATION}"
+
+tmp=$(mktemp -d)
+cd "${tmp}"
+
+# Create the index database from the bundle we just published
+opm index add \
+    --bundles "${BUNDLE_DESTINATION_DIGEST}" \
+    --out-dockerfile index.Dockerfile \
+    --generate
+
+# Add the index database to the opm base image
+# It does the same operation as in the Dockerfile generated by the opm index add... command above
+skopeo copy "docker://${OPM_BASE_IMAGE}" oci:opm:latest
+umoci unpack --rootless --image opm:latest bundle
+mv database bundle/rootfs/
+umoci repack --image opm:index bundle
+
+# Configure the index image and publish it
```

Comment:
```suggestion
```

Reply:
Just to make sure the comment above looks like it applies to the lines below as well

Reply:
Like that?

Reply:
yep